extensions/common/gcp/gcp-core/src/main/java/org/eclipse/edc/gcp/common/GcpCredentials.java

```diff
@@ -0,0 +1,133 @@
+     * otherwise it will be created based on the Application Default Credentials
+     *
+     * @return GoogleCredentials
+     */
+    public GoogleCredentials resolveGoogleCredentialsFromDataAddress(DataAddress dataAddress) {
+        if (dataAddress.getKeyName() != null && !dataAddress.getKeyName().isEmpty()) {
+            var tokenContent = vault.resolveSecret(dataAddress.getKeyName());
+            return createGoogleCredential(tokenContent, GcpCredentialType.GOOGLE_ACCESS_TOKEN);
+        } else if (dataAddress.getProperty(GcsStoreSchema.SERVICE_ACCOUNT_KEY_NAME) != null && !dataAddress.getProperty(GcsStoreSchema.SERVICE_ACCOUNT_KEY_NAME).isEmpty()) {
+            return createGoogleCredential(vault.resolveSecret(dataAddress.getProperty(GcsStoreSchema.SERVICE_ACCOUNT_KEY_NAME)), GcpCredentialType.GOOGLE_SERVICE_KEY);
+        } else if (dataAddress.getProperty(GcsStoreSchema.SERVICE_ACCOUNT_KEY_VALUE) != null && !dataAddress.getProperty(GcsStoreSchema.SERVICE_ACCOUNT_KEY_VALUE).isEmpty()) {
+            try {
+                var serviceKeyContent = new String(b64Decoder.decode(dataAddress.getProperty(GcsStoreSchema.SERVICE_ACCOUNT_KEY_VALUE)));
+                if (!serviceKeyContent.contains("service_account")) {
+                    throw new GcpException("SERVICE_ACCOUNT_VALUE is not provided as a valid service account key file.");
+                }
+                return createGoogleCredential(serviceKeyContent, GcpCredentialType.GOOGLE_SERVICE_KEY);
+            } catch (IllegalArgumentException ex) {
+                throw new GcpException("SERVICE_ACCOUNT_VALUE is not provided in a valid base64 format.");
+            }
+        } else {
+            return creatApplicationDefaultCredentials();
+        }
+    }
+
+    /**
+     * Returns the Google Credentials which will created based on the Application Default Credentials in the following approaches
+     * - Credentials file pointed to by the GOOGLE_APPLICATION_CREDENTIALS environment variable
+     * - Credentials provided by the Google Cloud SDK gcloud auth application-default login command
+     * - Google App Engine built-in credentials
+     * - Google Cloud Shell built-in credentials
+     * - Google Compute Engine built-in credentials
+     *
+     * @return GoogleCredentials
+     */
+    public GoogleCredentials creatApplicationDefaultCredentials() {
+        return createGoogleCredential("", GcpCredentialType.DEFAULT_APPLICATION);
+    }
+
+
+    public GoogleCredentials createGoogleCredential(String keyContent, GcpCredentialType gcpCredentialType) {
+        GoogleCredentials googleCredentials;
+
+        if (gcpCredentialType.equals(GcpCredentialType.GOOGLE_ACCESS_TOKEN)) {
+            try {
+                var gcpAccessToken = typeManager.readValue(keyContent, GcpAccessToken.class);
+                monitor.info("Gcp: The provided token will be used to resolve the google credentials.");
```

Comment:
If I remember it correct the log level in the rest of the extension in caeses like this was debug